xain_fl/grpc/coordinator.py

```diff
@@ -265,29 +277,41 @@ class Coordinator:
     def __init__(
         self,
         num_rounds: int = 10,
-        required_participants: int = 10,
-        aggregator: Optional[Aggregator] = None,
+        minimum_participants_in_round: int = 1,
+        fraction_of_participants: float = 1.0,
         theta: List[np.ndarray] = [],
         epochs: int = 0,
         epoch_base: int = 0,
     ) -> None:
-        self.required_participants = required_participants
+        self.minimum_participants_in_round = minimum_participants_in_round
+        self.fraction_of_participants = fraction_of_participants
         self.participants = Participants()
         self.num_rounds = num_rounds
-        self.aggregator = aggregator if aggregator else FederatedAveragingAgg()
+        self.aggregator = FederatedAveragingAgg()
+        self.controller = RandomController(self.participants.ids())
+        self.minimum_connected_participants = self.get_minimum_connected_participants()
 
         # global model
         self.theta = theta
         self.epochs = epochs
         self.epoch_base = epoch_base
 
         # round updates
-        self.round = Round(self.required_participants)
+        self.round = Round(self.participants.ids())
 
         # state variables
         self.state = coordinator_pb2.State.STANDBY
         self.current_round = 0
 
+    def get_minimum_connected_participants(self) -> float:
```

Comment:
Why float instead of int?

Reply:
It actually returns `int` as the return statement uses a floor division

Reply:
i'll make it obvious it's an `int`